src/execution/execute.js

```diff
@@ -565,19 +582,30 @@ function resolveField(
     rootValue: exeContext.rootValue,
     operation: exeContext.operation,
     variableValues: exeContext.variableValues,
+    executionPath: exePath
   };
 
   // Get the resolve function, regardless of if its result is normal
   // or abrupt (error).
   const result = resolveOrError(resolveFn, source, args, context, info);
 
-  return completeValueCatchingError(
+  const completed = completeValueCatchingError(
     exeContext,
     returnType,
     fieldASTs,
     info,
+    exePath,
     result
   );
+
+  if (isThenable(completed)) {
```

Comment:
This block shouldn't be necessary, since the `completeValueCatchingError` already does this work.